### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by neudabei**
Implemented registration for new users and login/logout for registered users.

### CODE REVIEW

File: app/views/sessions/new.html.haml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+%section.sign_in.container
+  .row
+    .col-sm-10.col-sm-offset-1
+      = bootstrap_form_tag url: '/login', layout: :horizontal, label_col: "col-sm-2", control_col: "col-sm-4" do |f|
```
Comment (tomtomecek) [inline app/views/sessions/new.html.haml]: I would not recommend hardcoding paths like `url: '/login'` - use Rails named routes for that purpose.

File: app/views/sessions/new.html.haml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        %form.sign_in
```
Comment (tomtomecek) [inline app/views/sessions/new.html.haml]: You are using `%form` html tag here - hard to believe this form works. You can not nest 2 forms.

Comment (tomtomecek) [inline app/views/sessions/new.html.haml]: In order to pass `.sign_in` to `bootstrap_form_for` you need to use `html: { class: "sign_in }` or something similar. Study the Rails API docks for form_for and bootstraps_form docs too.

File: app/views/shared/_header.html.haml [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
-            %a(href="#") Account
-            %a(href="#") Plan and Billing
-            %a(href="#") Sign Out
+    = link_to "MyFLiX", root_path
+  - if logged_in?
+    %ul.col-md-4.clearfix
+      %li= link_to "Videos", videos_path
+      %li= link_to "My Queue"
+      %li= link_to "People"
+    = form_tag search_videos_path, class: "span5 for-search", method: "get" do
```
Comment (tomtomecek) [inline app/views/shared/_header.html.haml]: This is Bootstrap 3 - `span5` is from Bootstrap 2 and will not work.

File: app/views/shared/_header.html.haml [hunk 4]
```diff
@@ -1,21 +1,22 @@
...
+      %input.search-query(name="search_term" type="text" placeholder="Search for videos here")
```
Comment (tomtomecek) [inline app/views/shared/_header.html.haml]: I would use Rails helpers here. `= text_field_tag :search_term, ...`

File: app/views/shared/_header.html.haml [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
+      %button.btn.btn-default(type="submit") Search
```
Comment (tomtomecek) [inline app/views/shared/_header.html.haml:11]: I would also use Rails helper here `= submit_tag ...`

File: app/views/users/new.html.haml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        /       %input.form-control(type="text")
+        /   .form-group
+        /     %label.control-label.col-sm-2 Expiration
+        /     .col-sm-3
+        /       = select_month(Date.today, {add_month_numbers: true},  class: 'form-control')
+        /     .col-sm-2
+        /       = select_year(Date.today.year, {start_year: Date.today.year, end_year: Date.today.year + 4}, class: 'form-control')
+        %fieldset.actions.control-group.col-sm-offset-2
+          .controls
+            %input(type="submit" value="Sign Up" class="btn btn-default")
```
Comment (tomtomecek) [inline app/views/users/new.html.haml:28]: I would use `= f.submit ...` here as well.

File: config/routes.rb [hunk 7]
```diff
@@ -10,4 +12,12 @@
 
   get '/genre/:id', to: 'categories#show'
 
+  # registration/authentication
+  get '/register', to: 'users#new'
+  get '/login', to: 'sessions#new'
+  post '/login', to: 'sessions#create'
+  get '/logout', to: 'sessions#destroy'
+
+  resources :users, only: [:show, :create, :edit, :update]
```
Comment (tomtomecek) [inline config/routes.rb]: Where are those methods `:show` and `:edit`, `:update` ?  So far you should not open new routes to stuff you have not built or secured.

**Comment by neudabei** [comment]
Thanks for your helpful comments, Tomas!

**Pull request merged**